docs/DeveloperGuide.md

```diff
@@ -125,7 +125,30 @@ that the UI automatically updates when the data in the list change.
 * Does not depend on any of the other three components.
 
 ### 3.5. State Component
-_{To Be Added}_
+ (Contributed by Song Yu)
+ 
+ ***API***: `State.java`
+ 
+ The `State` component:
+ * Stores a `BudgetIndex` instance that represents the budget that `NUSave` is currently displaying.
+ * Stores a `Page` enumeration that represents the type of `Page` NUSave is currently on. 
+ * Stores a `BooleanProperty` instance that triggers a callback on change of `boolean` value, updating the page title 
+ of the current page.
+ *  Stores a `String` value which represents the title of the current page.
+ 
+ `State` is used by `Model` to track the current state of NUSave. When a command that requires a different budget book
+ or list of budgets to be displayed, `State` will store information of the new page being displayed. This information
```

Comment:
Just a small change needed, I think the wording is slightly off here. Maybe you meant to say when the command is called or used.